renovate.json

```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "config:recommended"
+  ],
+  "packageRules": [
+    {
+      "enabled": false,
+      "packagePatterns": [
+        "*"
+      ]
+    },
+    {
+      "enabled": true,
+      "matchDepPatterns": [
+        "automattic|wpmreleasetoolkit"
+      ],
+      "separateMajorMinor": false
+    },
+    {
+      "enabled": true,
+      "enabledManagers": [
+        "gradle"
+      ],
+      "matchDepPatterns": [
+        "automattic"
+      ],
+      "separateMajorMinor": false,
+      "versioning": "semver-coerced"
+    }
```

Comment:
Question (❓): With this configuration now, we want to only enable dependency updates for anything that is of a `gradle` manager, which is also matching the `automattic` pattern, but also to only create one PR for a dependency update and is versioned as `semver-coerced`.
- What `gradle` manager means in this context? 🤔 
- Why using `semver-coerced` for this configuration, what's special about that? 🤔

This is what causing Renovate to create the below 1 (potential) PR, just because this is a `gradle` plugin, as if because of the `gradle` manager configuration, correct me if I am wrong:
- Update plugin `com.automattic.android.measure-builds` to v3

Reply:
Thanks for the questions!

> What gradle manager means in this context? 🤔

It means that this block of configuration is related to Gradle

> Why using semver-coerced for this configuration, what's special about that? 🤔

These internal discussions explains this matter in details: paaHJt-6tz-p2 paaHJt-6wb-p2 - TLDR; this is a main point of using Renovate over Dependabot, as we can specify this tool to create PRs only for meaningful versions (and not <pr>-<commit_hash> ones).

> This is what causing Renovate to create the below 1 

This actually causes Renovate to create all Gradle PRs with semver versions.

Reply:
> It means that this block of configuration is related to Gradle

Gradle, as if, everything `*.gradle` file related, or Gradle `plugins` blocks related, apologies if the question is too stupid, I am just trying to understand why we need to separate configurations, one with the `manager` and one without... 😅 

Why don't we use the `"versioning": "semver-coerced"` configuration on the above block, the one without the `manager`? 🤔

Reply:
No worries, good questions!

> Gradle, as if, everything *.gradle file related,

Everything Gradle related (also `.gradle.kts`, version catalogs etc.).

> Why don't we use the "versioning": "semver-coerced" configuration on the above block, the one without the manager? 🤔

Because the semver restriction is needed only for Gradle dependencies, as they release `<pr>-<commit hash>` versions. In the block above, where we whitelist e.g. "buildkite plugin automattic/a8c-ci-toolkit" I don't want to put semver restriction as it's not needed. It might also be invalid - I don't know what versioning strategies different projects might have.

Reply:
Gotcha, thanks for the clarification on all that Wojtek! 🥇 

> I don't know what versioning strategies different projects might have.

True, I don't know either... 🤷 

Well, we can always change the configuration later on if need be, so far so good. 👍